packages/apollo-gateway/src/index.ts

```diff
@@ -385,6 +411,33 @@ export class ApolloGateway implements GraphQLService {
     }
   }
 
+  private notifySchemaListeners(schema?: GraphQLSchema) {
+    try {
+      this.onSchemaChangeListeners.forEach(listener => listener(schema!));
+    } catch (e) {
+      this.logger.error(
+        "An error was thrown from an 'onSchemaChange' listener. " +
+        "The schema will still update: " + (e && e.message || e));
+    }
+  }
+
+  // This can be used without an argument in order to perform an ad-hoc health check
+  // of the downstream services like so:
+  // try {
+  //   await gateway.performServiceHealthChecks();
+  // } catch(e) {
+  //   /* your error handling here */
+  // }
+  protected performServiceHealthChecks(serviceMap: DataSourceMap = this.serviceMap) {
+    return Promise.all(
```

Comment:
I think this is probably the right approach to provide observability into _each_ services failure in the event of multiple, though I thought about suggesting `Promise.race` so that it would fail-fast.  Downside to `Promise.race`, is that it would only return the result of the first rejected/resolved.

Reply:
@trevor-scheer correctly noted offline that `Promise.race` is not a great suggestion here as successes would win that race from time to time.  `Promise.all` does pretty much what we want as it stands.

Reply:
Discussed offline, but determined this as it stands is the correct approach (though I understand your thought process!).

`Promise.all()` actually rejects as soon as one rejection occurs, meaning it's doing exactly what we'd want w.r.t. failing fast! For the success case, we actually _require_ the `all()` functionality since we want to wait for all services to respond.